### PR TITLE
Improved readability of the VIN model, in addition to minor changes

### PR DESCRIPTION
**Pull request by shuishida**
My main modification is in the forward method of the model where you extract the q_out from the q values, and not repeating q = F.conv2d(...) in two places. I also made minor improvements, such as adding argparse in the dataset creation script and changing .cuda() into .to(device) in test.py.

### CODE REVIEW

File: model.py [hunk 1]
```diff
@@ -37,32 +37,33 @@ def __init__(self, config):
...
-        q = self.q(r)
+    def forward(self, input_view, state_x, state_y, k):
+        """
+        :param input_view: (batch_sz, imsize, imsize)
+        :param state_x: (batch_sz,), 0 <= state_x < imsize
+        :param state_y: (batch_sz,), 0 <= state_y < imsize
+        :param k: number of iterations
+        :return: logits and softmaxed logits
+        """
+        h = self.h(input_view)  # intermediate output
```
Comment (kent-tri) [inline model.py]: nit: Please capitalize the first letter for the new comments

File: model.py [hunk 2]
```diff
@@ -37,32 +37,33 @@ def __init__(self, config):
...
+        r = self.r(h)           # reward output
+        q = self.q(r)           # initial Q value from reward for different actions
         v, _ = torch.max(q, dim=1, keepdim=True)
-        for i in range(0, config.k - 1):
+
+        for i in range(k):
```
Comment (kent-tri) [inline model.py]: This should be `range(k-1)` to stick with the paper

**Review by kent-tri — CHANGES_REQUESTED** [review]
Two notes, otherwise LGTM

File: model.py (hunk 1 above)
Comment (shuishida) [reply model.py]: That's a good point! Fixed it.

File: model.py (hunk 2 above)
Comment (shuishida) [reply model.py]: I see! I made a compromising commit by extracting the q evaluation step as a separate function.

**Comment by shuishida** [comment]
In addition to the fixes you suggested, I also refactored the `gridworld.py` file in the hope to make it more readable and concise.

**Comment by kentsommer** [comment]
LGTM

**Pull request merged**